coremltools/converters/mil/frontend/torch/ops.py

```diff
@@ -1650,34 +1650,84 @@ def pad(context, node):
     context.add(res)
 
 
+@register_torch_op
+def adaptive_avg_pool1d(context, node):
+    _adaptive_pool1d(context, node, mb.reduce_mean)
+
+
 @register_torch_op
 def adaptive_avg_pool2d(context, node):
     _adaptive_pool2d(context, node, mb.avg_pool, mb.reduce_mean)
 
 
+def _adaptive_pool1d(context, node, reduce_op):
+    inputs = _get_inputs(context, node, expected=2)
+    x = inputs[0]
+    assert len(inputs[1].val) == 1
+    out_length = inputs[1].val[0]
+
+    if len(x.shape) == 3:
+        # 3D input
+        begin_prefix = [0, 0]
+        end_prefix = [x.shape[0], x.shape[1]]
+        out_shape = [x.shape[0], x.shape[1], out_length]
+    else:
+        # 2D input
+        assert len(x.shape) == 2
+        begin_prefix = [0]
+        end_prefix = [x.shape[0]]
+        out_shape = [x.shape[0], out_length]
+
+    pool_results = []
+    for start, end in get_kernel_indexes_1d(x.shape[-1], out_length):
+        cur_kernel = mb.slice_by_index(
+            x=x,
+            begin=begin_prefix+[start],
+            end=end_prefix+[end],
+        )
+        cur_result = reduce_op(
+            x=cur_kernel,
+            axes=[-1],
+            keep_dims=True
+        )
+        pool_results.append(cur_result)
+        
+    context.add(
+        mb.reshape(
+            x=mb.concat(values=pool_results, axis=-1),
+            shape=out_shape,
+            name=node.name,
+        )
+    )
+
+
+@register_torch_op
+def adaptive_max_pool1d(context, node):
+    _adaptive_pool1d(context, node, mb.reduce_max)
+
+
 @register_torch_op
 def adaptive_max_pool2d(context, node):
     _adaptive_pool2d(context, node, mb.max_pool, mb.reduce_max)
 
 
+def get_kernel_indexes_1d(in_dimension, out_dimension):
```

Comment:
mind adding type hint for this function?

Reply:
Done